### PR TITLE
Implementing basic Cookie throwing system

### PR DESCRIPTION
**Pull request by nurgan**
* added CookieThrower, reading input for aiming and throwing
* added AimInputComponent for polling Input regarding the aiming

### CODE REVIEW

File: src/AimInputComponent.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
+#ifndef AIM_INPUT_COMPONENT_H
+#define AIM_INPUT_COMPONENT_H
+
+#include "InputComponent.h"
+
+class AimInputComponent : public InputComponent {
```
Comment (rgarmsen2295) [inline src/AimInputComponent.h:6]: I literally thought "this seems perfect for the input component" and then it already was, awesome lol

File: src/CookieThrower.cpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
+#include "CookieThrower.h"
+#include "GameManager.h"
+#include <glm/gtx/rotate_vector.hpp>
+
+/*
+ * Poll input from the given input handler, and throws a throwable
+ * dependent on the input and according constraints.
+ */
+CookieThrower::CookieThrower() {
```
Comment (rgarmsen2295) [inline src/CookieThrower.cpp:9]: I guess intuitively it makes the most sense for the drawing of the aiming line to be associated with the "thrower"... would also allow throwers to be attached to other objects and still see the line (for ai, etc.).  Just thinking out loud, not sure if that makes sense for the overall code.

File: src/CookieThrower.cpp [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    }
+    if(aimInputComponent->toggleYRotation){
+        yRot += aimInputComponent->velocity * deltaTime;
+    }
+
+    if (aimInputComponent->toggleThrow) {
+        if (totalTime >= previousCookieTime + 0.5) {
+
+            CookiePhysicsComponent *cookiePhysicsComp = new CookiePhysicsComponent();
+            BunnyRenderComponent *renderComp = new BunnyRenderComponent(cookieShape, progPhong, obsidian);
```
Comment (rgarmsen2295) [inline src/CookieThrower.cpp:39]: Should at some point have a "CookieRenderComponent" or maybe just have a "DefaultRenderComponent" for these sorts of objects where nothing special happens

Comment (joncatanio) [inline src/CookieThrower.cpp:39]: `progPhong` isn't being used like this anymore. Reed made a shader manager singleton, use `ShaderManager& shaderManager = ShaderManger::instance()` to get the shader manager and `shaderManager.getShaderProgram("Phong")` will get you the phong shader `Program`.

File: src/GameManager.h [hunk 4]
```diff
@@ -1,9 +1,6 @@
 #ifndef GAME_MANAGER_H
 #define GAME_MANAGER_H
 
-#include <memory>
-#include <string>
```
Comment (rgarmsen2295) [inline src/GameManager.h:5]: Good clean-up, thanks!

**Review by rgarmsen2295 — APPROVED** [review]
+1 looks good, left a few comments

File: src/CookieThrower.cpp (hunk 3 above)
Comment (nurgan) [reply src/CookieThrower.cpp:39]: yeah, I just reused the BunnyComponent for now, I think we should just rename the BunnyRenderComponent to PhongRenderComponent

File: src/CookieThrower.cpp (hunk 2 above)
Comment (nurgan) [reply src/CookieThrower.cpp:9]: makes sense to me.

File: src/CookieThrower.cpp [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+                    GameObjectType::NONSTATIC_OBJECT,
+                    player.getPosition(),
+                    throwDirection,
+                    startVelocity,
+                    initialScale,
+                    NULL,
+                    cookiePhysicsComp,
+                    renderComp);
+
+            gameManager.getGameWorld().addNonStaticGameObject(cookieObj);
```
Comment (joncatanio) [inline src/CookieThrower.cpp:55]: we're gonna want to change this to `addDynamicGameObject`, see #28 for more info.

File: src/GameWorld.cpp [hunk 6]
```diff
@@ -40,21 +42,10 @@ void GameWorld::resetWorld() {
 void GameWorld::updateGameObjects(double deltaTime, double totalTime) {
 	// Keep track of the last spawn time internally to know when to spawn next
 	static double previousSpawnTime = 0.0;
-    static double previousCookieTime = 0.0;
 
-	// Spawn a new bunny every ~3 seconds, and max out at 30 bunnies
```
Comment (joncatanio) [inline src/GameWorld.cpp:45]: Why no more bunnies? We should leave this code in to ensure the player's bounding box is correctly functioning.

Comment (nurgan) [reply src/GameWorld.cpp:45]: oh shit, I removed that when refactoring my code a little to fast. I'll bring it back

**Pull request merged**